akka-parsing/src/main/scala/akka/parboiled2/CharPredicate.scala

```diff
@@ -88,15 +93,16 @@ sealed abstract class CharPredicate extends (Char => Boolean) {
       case ix => Some(string charAt ix)
     }
 
-  protected def or(that: Char => Boolean): CharPredicate =
-    from(if (this == Empty) that else c => this(c) || that(c))
-  protected def and(that: Char => Boolean): CharPredicate =
+  protected def or(that: CharPredicate): CharPredicate =
+    if (this == Empty) that else from(c => this(c) || that(c))
+  protected def and(that: CharPredicate): CharPredicate =
     if (this == Empty) Empty else from(c => this(c) && that(c))
-  protected def andNot(that: Char => Boolean): CharPredicate =
-    from(if (this == Empty) c => !that(c) else c => this(c) && !that(c))
+  protected def andNot(that: CharPredicate): CharPredicate =
+    if (this == Empty) from(c => !that(c)) else from(c => this(c) && !that(c))
 }
 
-object CharPredicate {
+@InternalApi
```

Comment:
Added that for completeness. The whole akka-parsing module does not provide binary compatibility, though.